src/System.CommandLine.Subsystems/ValueSources/AggregateValueSource.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine.ValueSources;
+
+public class AggregateValueSource<T> : ValueSource<T>
+{
+    private List<ValueSource<T>> valueSources = [];
+
+    public AggregateValueSource(ValueSource<T> firstSource,
```

Comment:
More doc comments will be needed throughout this project. Add TODO or issue or fix